storage/factory/latestDataProvider.go

```diff
@@ -0,0 +1,227 @@
+	return &latestDataProvider{
+		generalConfig:         args.GeneralConfig,
+		marshalizer:           args.Marshalizer,
+		hasher:                args.Hasher,
+		workingDir:            args.WorkingDir,
+		chainID:               args.ChainID,
+		directoryReader:       args.DirectoryReader,
+		defaultShardString:    args.DefaultShardString,
+		defaultEpochString:    args.DefaultEpochString,
+		defaultDBPath:         args.DefaultDBPath,
+		bootstrapDataProvider: args.BootstrapDataProvider,
+	}, nil
+}
+
+// Get will return a struct containing the latest data in storage
+func (ldp *latestDataProvider) Get() (storage.LatestDataFromStorage, error) {
+	parentDir, lastEpoch, err := ldp.GetParentDirAndLastEpoch()
+	if err != nil {
+		return storage.LatestDataFromStorage{}, err
+	}
+
+	return ldp.getLastEpochAndRoundFromStorage(parentDir, lastEpoch)
+}
+
+// GetParentDirAndLastEpoch returns the parent directory and last epoch
+func (ldp *latestDataProvider) GetParentDirAndLastEpoch() (string, uint32, error) {
+	parentDir := filepath.Join(
+		ldp.workingDir,
+		ldp.defaultDBPath,
+		ldp.chainID)
+
+	directoriesNames, err := ldp.directoryReader.ListDirectoriesAsString(parentDir)
+	if err != nil {
+		return "", 0, err
+	}
+
+	epochDirs := make([]string, 0, len(directoriesNames))
+	for _, dirName := range directoriesNames {
+		isEpochDir := strings.HasPrefix(dirName, ldp.defaultEpochString)
+		if !isEpochDir {
+			continue
+		}
+
+		epochDirs = append(epochDirs, dirName)
+	}
+
+	lastEpoch, err := ldp.GetLastEpochFromDirNames(epochDirs)
+	if err != nil {
+		return "", 0, err
+	}
+
+	return parentDir, lastEpoch, nil
+}
+
+func (ldp *latestDataProvider) getLastEpochAndRoundFromStorage(parentDir string, lastEpoch uint32) (storage.LatestDataFromStorage, error) {
+	persisterFactory := NewPersisterFactory(ldp.generalConfig.BootstrapStorage.DB)
+	pathWithoutShard := filepath.Join(
+		parentDir,
+		fmt.Sprintf("%s_%d", ldp.defaultEpochString, lastEpoch),
+	)
+	shardIdsStr, err := ldp.GetShardsFromDirectory(pathWithoutShard)
+	if err != nil {
+		return storage.LatestDataFromStorage{}, err
+	}
+
+	var mostRecentBootstrapData *bootstrapStorage.BootstrapData
+	var mostRecentShard string
+	highestRoundInStoredShards := int64(0)
+	epochStartRound := uint64(0)
+
+	for _, shardIdStr := range shardIdsStr {
+		persisterPath := filepath.Join(
+			pathWithoutShard,
+			fmt.Sprintf("%s_%s", ldp.defaultShardString, shardIdStr),
+			ldp.generalConfig.BootstrapStorage.DB.FilePath,
+		)
+
+		bootstrapData, storer, errGet := ldp.bootstrapDataProvider.LoadForPath(persisterFactory, persisterPath)
+		if errGet != nil {
+			continue
+		}
+
+		if bootstrapData.LastRound > highestRoundInStoredShards {
+			shardID := uint32(0)
+			var err error
+			shardID, err = convertShardIDToUint32(shardIdStr)
+			if err != nil {
+				continue
+			}
+			epochStartRound, err = loadEpochStartRound(shardID, bootstrapData.EpochStartTriggerConfigKey, storer)
```

Comment:
this function is used only here, so it should be moved from common to here.

Reply:
done.